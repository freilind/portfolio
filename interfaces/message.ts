export interface IMessage {
    es: { [key: string]: any; };
    en: { [key: string]: any; };
    [key: string]: any;
}

