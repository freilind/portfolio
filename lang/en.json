{
    "description.name": "Freilin Manzano",
    "description.greeting": "Dear (a), first of all a cordial greeting.",
    "description.body.part1": "My name is Freilin Manzano, I was born in Venezuela",
    "description.body.part2": " and I am currently based in Chile",
    "description.body.part3": " I am dedicated to web development, with experience (+8 years) and Java certification (Java SE 8 Programmer I 1Z0-808), enthusiastic about the front-end area with React.js, Next.js, TailwindCss, Flutter.",

    "collapse.title.1": "Portfolio",
    "collapse.subtitle.1": "Projects made  with react.",
    "collapse.title.2": "Experience",
    "collapse.subtitle.2": "Work experience.",
    "collapse.title.3": "Education",
    "collapse.subtitle.3": " ",
    "collapse.title.4": "Certifications & Courses",
    "collapse.subtitle.4": " ",

    "experienceList": [
        {
            "company": "Nisum Latam",
            "date": "May 16 - Present",
            "description": "Development of a mortgage loan product for the banking sector. Developing new components for microservices architecture and maintaining the integration with the monolith for the continuity of the operation. Supporting the organization in digital transformation and agile methodologies.",
            "tech": "Technologies: java, spring boot, spring cloud, angular, nodejs, jenkins.",
            "rol": "Senior Developer"
        },
        {
            "company": "LOREBI",
            "date": "Jan 15 - Apr 16",
            "description": "Development of a free Wi-Fi platform that requires the person to watch segmented advertising before allowing them to browse for a certain time. My role was the development of the entire application using java and the Spring mvc framework. Development of a dashboard that displays the metrics and behavior of Wi-Fi users.",
            "tech": "Technologies: java, spring mvc, jpa, hibernate, glassfish, maven, svn.",
            "rol": "IT Specialist"
        },
        {
            "company": "Corp. Wimac Systems",
            "date": "Nov 13 - May 14",
            "description": "Project on claims and massive failures. Telecommunications Movilnet C.A. Module development to record customer complaints and generate statistics and reports that will help business management to make decisions regarding failures. My role was the development of the view and controller layer of the system.",
            "tech": "Technologies: java, primefaces, ejb, Jboss, python, eclipseIDE, postgresql.",
            "rol": "IT Specialist"
        },
        {
            "company": "Eliexir Cm Computer",
            "date": "Nov 12 - Nov 13",
            "description": "Voice Project module for CVSC+. Telefonica Venezuela. Development of a module to associate telephone numbers in order to achieve direct communication with special plans and conditions. My role was the development of a webservice that would use the corresponding stored procedures for the activation and deactivation of plans and services in the voice VPN module.",
            "tech": "Technologies: java, weblogic 8.5, web services, eclipse IDE, svn, maven soap UI.",
            "rol": "Java developer"
        }
    ],

    "educationHead": [
        {
            "key": "organization",
            "label": "ORGANIZATION"
        },
        {
            "key": "description",
            "label": "DESCRIPTION"
        },
        {
            "key": "date",
            "label": "DATE"
        }
    ],

    "educationList": [
        {
            "key": "1",
            "organization": "ORACLE",
            "description": "Oracle Certified Associate, Java SE 8 Programmer",
            "date": "2019"
        },
        {
            "key": "2",
            "organization": "UNITEC",
            "description": "Diplomat BPM-SOA",
            "date": "2016"
        },
        {
            "key": "3",
            "organization": "IUGT",
            "description": "Higher University Computer Technician",
            "date": "2012"
        }
    ]
}