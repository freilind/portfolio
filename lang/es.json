{
    "description.name": "Freilin Manzano",
    "description.greeting": "Estimado (a), ante todo un coordial saludo.",
    "description.body.part1": "Mi nombre es Freilin Manzano, nací en Venezuela",
    "description.body.part2": " y actualmente estoy radicado en Chile",
    "description.body.part3": " Me dedico al Desarrollo web, con experiencia (+8 años) y certificación Java (Java SE 8 Programmer I 1Z0-808), entusiasta por el área de front-end con React.js, Next.js, TailwindCss, Flutter.",

    "collapse.title.1": "Portafolio",
    "collapse.subtitle.1": "Proyectos de react.",
    "collapse.title.2": "Experiencia",
    "collapse.subtitle.2": "Experiencia laboral.",
    "collapse.title.3": "Educación",
    "collapse.subtitle.3": " ",
    "collapse.title.4": "Certificaciones & Cursos",
    "collapse.subtitle.4": " ",

    "experienceList": [
        {
            "company": "Nisum Latam",
            "date": "May 16 - Actualidad",
            "description": "Desarrollo de producto crédito hipotecario para el sector bancario. Desarrollando nuevos componentes para arquitectura de microservicios y mantener la integración con el monolito para la continuidad de la operación. Apoyando a la organización en la transformación digital y metodologías ágiles.",
            "tech": "Tecnologías: java, spring boot, spring cloud, angular, nodejs, jenkins.",
            "rol": "Senior Developer"
        },
        {
            "company": "LOREBI",
            "date": "Ene 15 - Abr 16",
            "description": "Desarrollo de una plataforma de wifi gratuita que requiere que la persona observe publicidad segmentada antes de permitirle navegar por un tiempo determinado. Mi función era el desarrollo de toda la aplicación utilizando java y framework Spring mvc. Desarrollo de un dashboard que donde se visualicen las métricas y comportamiento de los usuarios del wifi.",
            "tech": "Tecnologías: java , spring mvc, jpa, hibernate, glassfish, maven, svn.",
            "rol": "Especialista IT"
        },
        {
            "company": "Corp. Wimac Systems",
            "date": "Nov 13 - May 14",
            "description": "Proyecto sobre reclamos y fallas masivas. Telecomunicaciones Movilnet C.A. Desarrollo de módulo para registrar los reclamos de los clientes y generar estadísticas y reportes que ayudaran a la gerencia del negocio a tomar decisiones con respecto a las fallas. Mi función era el desarrollo de la capa vista y controlador del sistema.",
            "tech": "Tecnologías: java, primefaces, ejb, Jboss, python, eclipseIDE, postgresql.",
            "rol": "Especialista IT"
        },
        {
            "company": "Eliexir Cm Computer",
            "date": "Nov 12 - Nov 13",
            "description": "Proyecto de Voz módulo para CVSC+. Telefónica Venezuela. Desarrollo de módulo para asociar números telefónicos con la finalidad de lograr la comunicación directa con planes y condiciones especiales. Mi función era el desarrollo de webservice que utilizaran los stored procedured corespondientes para la activación y desactivación de planes y servicios en el módulo de VPN de voz.",
            "tech": "Tecnologías: java, weblogic 8.5, web services, eclipse IDE, svn, maven soap UI.",
            "rol": "Desarrollador Java"
        }
    ],

    "educationHead": [
        {
            "key": "organization",
            "label": "ORGANIZACIÓN"
        },
        {
            "key": "description",
            "label": "DESCRIPCIÓN"
        },
        {
            "key": "date",
            "label": "FECHA"
        }
    ],

    "educationList": [
        {
            "key": "1",
            "organization": "ORACLE",
            "description": "Oracle Certified Associate, Java SE 8 Programmer",
            "date": "2019"
        },
        {
            "key": "2",
            "organization": "UNITEC",
            "description": "Diplomado BPM-SOA",
            "date": "2016"
        },
        {
            "key": "3",
            "organization": "IUGT",
            "description": "Técnico Superior Universitario Informática",
            "date": "2012"
        }
    ]
}