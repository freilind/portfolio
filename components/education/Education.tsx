import React, { FC } from "react";
import { Grid, Table } from "@nextui-org/react";
import en from "../../lang/en.json";
import es from "../../lang/es.json";
import { useRouter } from "next/router";
import { IColumn, IItem, IMessage } from "../../interfaces";

const languages: IMessage = {
  es,
  en,
};

const Education: FC = () => {
  const router = useRouter();
  const locale = router.locale || "es";
  const { educationList, educationHead } = languages[locale];

  return (
    <Grid.Container gap={2} justify="center" xs={10}>
      <Table
        aria-label="Table of education"
        bordered={true}
        css={{
          height: "auto",
          minWidth: "100%",
        }}
      >
        <Table.Header columns={educationHead}>
          {(column: IColumn) => (
            <Table.Column
              key={column.key}
              aria-labelledby={`column-${column.key}`}
            >
              {column.label}
            </Table.Column>
          )}
        </Table.Header>
        <Table.Body items={educationList}>
          {(item: IItem) => (
            <Table.Row key={item.key} aria-labelledby={`row-${item.key}`}>
              {(columnKey) => (
                <Table.Cell
                  aria-labelledby={`cell-${item.key}-${item[columnKey]}`}
                >
                  {item[columnKey]}
                </Table.Cell>
              )}
            </Table.Row>
          )}
        </Table.Body>
      </Table>
    </Grid.Container>
  );
};

export default Education;
