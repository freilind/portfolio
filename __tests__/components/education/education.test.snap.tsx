import { FC } from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import es from "../../../lang/es.json";
import en from "../../../lang/en.json";
import { RouterContext } from "next/dist/shared/lib/router-context";
import { createMockRouter } from "../../../__test-utils__/createMockRouter";
import { IMessage } from "@interfaces/message";
import IntlProvider from "react-intl/src/components/provider";
import Education from "../../../components/education/Education";

const messages: IMessage = {
  es,
  en,
};
const lang = "es";

const IntlWrapper: FC = ({ children }: { children?: React.ReactNode }) => (
  <IntlProvider locale={lang} messages={messages[lang]}>
    {children}
  </IntlProvider>
);

describe("Education snapshot", () => {
  it.skip("renders education unchanged", () => {
    const { container } = render(
      <RouterContext.Provider value={createMockRouter({ route: lang })}>
        <Education />
      </RouterContext.Provider>,
      { wrapper: IntlWrapper }
    );
    expect(container).toMatchSnapshot();
  });
});
