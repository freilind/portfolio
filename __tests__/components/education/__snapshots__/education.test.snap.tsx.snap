// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Education snapshot renders education unchanged 1`] = `
<div>
  <div
    class="nextui-c-kRHeuF nextui-c-kRHeuF-iDbpjR-css nextui-grid-item xs nextui-grid-container"
  >
    <div
      class="nextui-c-hWZRae nextui-c-hWZRae-gikTHb-shadow-true nextui-c-hWZRae-kUADSW-bordered-true nextui-c-hWZRae-bzEnZV-borderWeight-normal nextui-c-hWZRae-iPJLV-css nextui-table-container"
    >
      <table
        aria-describedby=""
        aria-label="Table of education"
        class="nextui-c-cuoPNV nextui-c-cuoPNV-fyxmJu-color-primary nextui-c-cuoPNV-fPgLZC-lineWeight-light nextui-c-cuoPNV-ixxsp-css nextui-table"
        id="react-aria9466982254-2"
        role="grid"
        tabindex="0"
      >
        <thead
          class="nextui-c-PJLV nextui-table-row-group"
          role="rowgroup"
        >
          <tr
            class="nextui-c-fRbcxI nextui-table-header-row"
            role="row"
          >
            <th
              aria-labelledby="column-organization"
              class="nextui-c-cpCRyO nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
              data-key="organization"
              id="react-aria9466982254-2-organization"
              role="columnheader"
              tabindex="-1"
            >
              ORGANIZACIÓN
            </th>
            <th
              aria-labelledby="column-description"
              class="nextui-c-cpCRyO nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
              data-key="description"
              id="react-aria9466982254-2-description"
              role="columnheader"
              tabindex="-1"
            >
              DESCRIPCIÓN
            </th>
            <th
              aria-labelledby="column-date"
              class="nextui-c-cpCRyO nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-column-header"
              data-key="date"
              id="react-aria9466982254-2-date"
              role="columnheader"
              tabindex="-1"
            >
              FECHA
            </th>
          </tr>
          <tr
            aria-hidden="true"
            class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-icCJVMZ-css nextui-table-hidden-row"
          />
        </thead>
        <tbody
          class="nextui-c-PJLV nextui-c-PJLV-iccTTUn-css nextui-table-body"
          role="rowgroup"
        >
          <tr
            aria-hidden="false"
            aria-labelledby="react-aria9466982254-2-1-organization"
            class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
            data-key="1"
            role="row"
            tabindex="-1"
          >
            <td
              class="nextui-c-PJLV nextui-c-coKxkr nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
              data-key="11organization"
              id="react-aria9466982254-2-1-organization"
              role="rowheader"
              tabindex="-1"
            >
              ORACLE
            </td>
            <td
              class="nextui-c-PJLV nextui-c-coKxkr nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
              data-key="11description"
              role="gridcell"
              tabindex="-1"
            >
              Oracle Certified Associate, Java SE 8 Programmer
            </td>
            <td
              class="nextui-c-PJLV nextui-c-coKxkr nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
              data-key="11date"
              role="gridcell"
              tabindex="-1"
            >
              2019
            </td>
          </tr>
          <tr
            aria-hidden="false"
            aria-labelledby="react-aria9466982254-2-2-organization"
            class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
            data-key="2"
            role="row"
            tabindex="-1"
          >
            <td
              class="nextui-c-PJLV nextui-c-coKxkr nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
              data-key="22organization"
              id="react-aria9466982254-2-2-organization"
              role="rowheader"
              tabindex="-1"
            >
              UNITEC
            </td>
            <td
              class="nextui-c-PJLV nextui-c-coKxkr nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
              data-key="22description"
              role="gridcell"
              tabindex="-1"
            >
              Diplomado BPM-SOA
            </td>
            <td
              class="nextui-c-PJLV nextui-c-coKxkr nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
              data-key="22date"
              role="gridcell"
              tabindex="-1"
            >
              2016
            </td>
          </tr>
          <tr
            aria-hidden="false"
            aria-labelledby="react-aria9466982254-2-3-organization"
            class="nextui-c-bUbfzy nextui-c-PJLV nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-row"
            data-key="3"
            role="row"
            tabindex="-1"
          >
            <td
              class="nextui-c-PJLV nextui-c-coKxkr nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
              data-key="33organization"
              id="react-aria9466982254-2-3-organization"
              role="rowheader"
              tabindex="-1"
            >
              IUGT
            </td>
            <td
              class="nextui-c-PJLV nextui-c-coKxkr nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
              data-key="33description"
              role="gridcell"
              tabindex="-1"
            >
              Técnico Superior Universitario Informática
            </td>
            <td
              class="nextui-c-PJLV nextui-c-coKxkr nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-table-cell"
              data-key="33date"
              role="gridcell"
              tabindex="-1"
            >
              2012
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
